Remove configuration options for direct TCP replication. (#13647)
matrix-org/synapse: docs/upgrade.md

    dpkg -i matrix-synapse-py3_1.3.0+stretch1_amd64.deb
    ```

# Upgrading to v1.67.0

## Direct TCP replication is no longer supported: migrate to Redis

Redis support was added in v1.13.0 with it becoming the recommended method in
v1.18.0. It replaced the old direct TCP connections (which was deprecated as of
v1.18.0) to the main process. With Redis, rather than all the workers connecting
to the main process, all the workers and the main process connect to Redis,
which relays replication commands between processes. This can give a significant

> cpu saving on the main process and will be a prerequisite for upcoming
> performance improvements.

Reviewer: ```suggestion
CPU saving on the main process and is a prerequisite for upcoming
performance improvements.
```

Author: Thanks! I copied this from the docs and didn't proof it too hard. 👍

To migrate to Redis add the [`redis` config](./workers.md#shared-configuration),

> and remove the TCP `replication` listener from master and `worker_replication_port`

Reviewer: I don't know what the TCP configuration looked like, but I think that you still need the `replication` listener in the `listeners` block of the config, in order to serve the portions of replication that happen over HTTP requests?
This paragraph risks misleading me into removing the HTTP listener. Think it would be worth clarifying if you know how?

Edit: I think I see. You still want the `replication` *resource* on the `type: http` listener, but there was previously a `type: replication` listener.

Author: Thanks, @turt2live had the same feedback and I failed to mark that down here. 👍

Author: I updated this a bit in 398c493d68f71a5b4a75c3839ea559aeeebd51ae, do you think it is clearer now?

Reviewer: Yep, that's good — thanks!
